### PR TITLE
Fix collecting events in cairo0

### LINKED ISSUE #1633: Fix collecting events in cairo0
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

In forking mode, we don't catch the events from cairo0 contracts - needs to be fixed

### PR DESCRIPTION
**Pull request by Draggu**
Closes #1633

## Introduced changes

- Fixes events collecting in cairo0

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/syscall_hooks.rs [hunk 1]
```diff
@@ -14,12 +17,38 @@ pub fn emit_event_hook(
         .code_address
         .unwrap_or(syscall_handler.call.storage_address);
```
[the remark this answers is not in the document]
Comment (piotmag769) [reply crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/syscall_hooks.rs:18]: cc @Arcticae as Marcin is off today

**Review by war-in — COMMENTED** [review]
We need to have an integration test for that (to see how it works on Cairo lvl).
Also it will be necessary to mention in docs that users will be able to only assert manually ([assert_emitted](https://foundry-rs.github.io/starknet-foundry/appendix/cheatcodes/spy_events.html#simple-one-in-which-user-asserts-with-assert_emitted-method) will  (probably) not work for cairo 0 events)

File: crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/syscall_hooks.rs [hunk 2]
```diff
@@ -1,25 +1,50 @@
...
     runtime_extensions::forge_runtime_extension::cheatcodes::spy_events::Event,
     state::CheatnetState,
 };
+use blockifier::execution::{
+    call_info::OrderedEvent, deprecated_syscalls::hint_processor::DeprecatedSyscallHintProcessor,
+    syscalls::hint_processor::SyscallHintProcessor,
+};
+use starknet_api::core::ContractAddress;
+
+pub trait SyscallHintProcessorExt {
```
[the remark this answers is not in the document]
Comment (Draggu) [reply crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/syscall_hooks.rs:11]: Going thru our codebase we use `Extension` in context of `ExtendedRuntime` and `Ext` for extension traits. This seems fine to me, but we can change it if you see some reason for this.

**Comment by Draggu** [comment]
> We need to have an integration test for that (to see how it works on Cairo lvl). Also it will be necessary to mention in docs that users will be able to only assert manually ([assert_emitted](https://foundry-rs.github.io/starknet-foundry/appendix/cheatcodes/spy_events.html#simple-one-in-which-user-asserts-with-assert_emitted-method) will (probably) not work for cairo 0 events)

Actually [assert_emitted](https://foundry-rs.github.io/starknet-foundry/appendix/cheatcodes/spy_events.html#simple-one-in-which-user-asserts-with-assert_emitted-method) works with cairo0, we only need to declare event emitted from cairo0 contract

```rs
#[starknet::contract]
mod _Cairo0Events {
    use starknet::ContractAddress;

    #[storage]
    struct Storage {}

    #[event]
    #[derive(Drop, starknet::Event)]
    enum Event {
        Cairo0EventName: Cairo0EventType,
    }

    #[derive(Drop, starknet::Event)]
    struct Cairo0EventType {
        some_data: felt252
    }
}
```
and then use it

```rs
spy.assert_emitted(@array![
    (
        cairo0_contract_address,
        _Cairo0Events::Event::Cairo0EventName(
            _Cairo0Events::Cairo0EventType {
                some_data: 123456789
            }
        )
    )
]);
```

File: crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/syscall_hooks.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (war-in) [reply crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/syscall_hooks.rs:11]: Ok, if you've checked that and it's aligned with what we have I'm ok with it

**Review by Arcticae — APPROVED** [review]

File: crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/syscall_hooks.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (piotmag769) [reply crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/syscall_hooks.rs]: We should change it imo
@Arcticae @war-in @Draggu

**Pull request merged**